contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.txt

```diff
@@ -0,0 +1,252 @@
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <mpi.h>
+
+#define P2P_LOG_MSG "p2p_log_%d.txt"
+#define P2P_LOG_REQUEST "p2p_log_request_%d.txt"
+
+// This is an array in the order of the calls to MPI_Recv/MPI_Irecv
+struct p2p_log_msg {
+  int count;
+  MPI_Datatype datatype;
+  int source;
+  int tag;
+  int MPI_ERROR;
+  MPI_Comm comm; // comm == MPI_COMM_NULL if this logs a failed MPI_Iprobe
+  MPI_Request request;
+};
+struct p2p_log_request {
+  MPI_Request request;
+  int source;
+  int tag;
+};
+
+void p2p_log(int count, MPI_Datatype datatype, int source, int tag,
+             MPI_Comm comm, MPI_Status *status, MPI_Request *request);
+void get_next_msg(struct p2p_log_msg *p2p_msg);
+int iprobe_next_msg(struct p2p_log_msg *p2p_msg);
+void set_next_msg(int count, MPI_Datatype datatype,
+                  int source, int tag, MPI_Comm comm,
+                  MPI_Status *status, MPI_Request *request);
```

Comment:
Please add extern "C" to these functions since the header file is used by both C++ and C file. Otherwise, the libmana.so can not find the symbol when testing vasp.
`diff --git a/contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.txt b/contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.txt
index 62024f1a..bf3be960 100644
--- a/contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.txt
+++ b/contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.txt
@@ -27,13 +27,19 @@ struct p2p_log_request {
   int tag;
 };
 
-void p2p_log(int count, MPI_Datatype datatype, int source, int tag,
-             MPI_Comm comm, MPI_Status *status, MPI_Request *request);
-void get_next_msg(struct p2p_log_msg *p2p_msg);
-int iprobe_next_msg(struct p2p_log_msg *p2p_msg);
-void set_next_msg(int count, MPI_Datatype datatype,
-                  int source, int tag, MPI_Comm comm,
-                  MPI_Status *status, MPI_Request *request);
+#if defined(__cplusplus)
+extern "C" {
+#endif
+  void p2p_log(int count, MPI_Datatype datatype, int source, int tag,
+               MPI_Comm comm, MPI_Status *status, MPI_Request *request);
+  void get_next_msg(struct p2p_log_msg *p2p_msg);
+  int iprobe_next_msg(struct p2p_log_msg *p2p_msg);
+  void set_next_msg(int count, MPI_Datatype datatype,
+                    int source, int tag, MPI_Comm comm,
+                    MPI_Status *status, MPI_Request *request);
+#if defined(__cplusplus)
+}
+#endif
 
`

Reply:
Great catch.  Thanks.

I've added the
```
#if defined(__cplusplus)
extern "C" {
#endif
...
#if defined(__cplusplus)
}
#endif
```
 around the entire p2p-deterministic.txt, which generates `p2p-deterministic.h`.  It's in the spirit of your suggested fix.